notebooks/6_search_in_continuous_spaces/index.md

```diff
@@ -0,0 +1,282 @@
+
+# Convexity
+A set C is convex if for any two points x,y ∈C ,αx+1-αy∈C for all α∈0,1, i.e., all points on the line between x and y also lie in C. A function f is convex if its domain C is convex and
+
+fαx+1-αy≤αfx+1-αfy (\*)
+
+*Figure  SEQ Figure \\* ARABIC 3. In convex function f, for every two point x,y∈domainf, the line segment between them lies above the graph of f.*
+![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.005.png)![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.006.jpeg)for all α∈0,1, i.e., the line segment betweenx,fxandy,fy, which is the chord from x to y, lies above the graph of f.
+
+
+For an affine function we always have equality in (\*), so all affine (and therefore also linear) functions are both convex and concave. Conversely, any function that is convex and concave is affine.
+
+A function is convex if and only if it is convex when restricted to any line that intersects its domain. In other words, f is convex if and only if for all x ∈ domain (f) and all v, the function gt=fx+tv is convex (on its domain, {t | x + t v ∈ domain (f) }). This property is very useful, since it allows us to check whether a function is convex by restricting it to a line.
+
+*Figure  SEQ Figure \\* ARABIC 4. A Non-convex function vs a Convex function*
+
+![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.007.png)![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.008.png)
+## Convex versus non-Convex Optimization Problems
+*Figure  SEQ Figure \\* ARABIC 5. convex region*
+![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.009.png)![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.010.png)A *convex optimization problem* is a problem where all of the constraints are convex functions, and the objective is a convex function if minimizing, or a concave function if maximizing. In a convex optimization problem, the feasible region is a convex region, as pictured below.
+
+With a convex objective and a convex feasible region, there can be only one optimal solution, which is globally optimal.  Several methods will either find the globally optimal solution, or prove that there is no feasible solution to the problem.  Convex problems can be solved efficiently up to very large size.
+
+*Figure  SEQ Figure \\* ARABIC 6. Non-convex region*
+
+![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.011.png)![](Aspose.Words.f2571b5b-5e5e-42a8-b357-02f03b593c59.012.png)A *non-convex optimization problem* is any problem where the objective or any of the constraints are non-convex, as pictured below.
+
+Such a problem may have multiple feasible regions and multiple locally optimal points within each region.  It can take time exponential in the number of variables and constraints to determine that a non-convex problem is infeasible, that the objective function is unbounded, or that an optimal solution is the "global optimum" across all feasible regions.
+##
+## Local and global optimization:
+### Local optimization 

```

Comment:
It's better to add examples or some figures which can show the difference between these two concepts